e2e_tests/conftest.py

```diff
@@ -22,21 +27,49 @@ def pytest_sessionstart(session: pytest.Session) -> None:
 
     server = environ.get("E2E_ENVIRONMENT")
     api_key = environ.get("E2E_API_KEY")
+    team_slug = environ.get("E2E_TEAM")
 
     if server is None:
         raise E2EEnvironmentVariableNotSet("E2E_ENVIRONMENT")
 
     if api_key is None:
         raise E2EEnvironmentVariableNotSet("E2E_API_KEY")
 
+    if team_slug is None:
+        raise E2EEnvironmentVariableNotSet("E2E_TEAM")
+
     if not isinstance(session.config.cache, pytest.Cache):
         raise TypeError("Pytest caching is not enabled, but E2E tests require it")
 
     session.config.cache.set("server", server)
     session.config.cache.set("api_key", api_key)
+    session.config.cache.set("team_slug", team_slug)
+
+    global datasets
+    datasets = setup_tests(ConfigValues(server=server, api_key=api_key, team_slug=team_slug))
+
+    print("Sleeping for 10 seconds to allow the server to catch up")
+    sleep(10)
```

Comment:
Sleep is also usually bad, but for test setup to account for latency, it's that or polling.